File: data/src/main/java/com/tonitealive/app/data/cache/UserCache.java
package com.tonitealive.app.data.cache;

import com.google.common.base.Optional;
import com.tonitealive.app.domain.model.User;


public interface UserCache {

    /**
     * Puts the current user in the cache
     *
     * @param user The user
     */
    void putCurrentUser(User user);

    /**
     * Retrieves the current user from the cache
     *
     * @return The cached user or null
     */
    Optional<User> getCurrentUser();

    /**
     * Clears the current user from the cache
     */
    void clearCurrentUser();

}

File: data/src/main/kotlin/com/tonitealive/app/data/UserCache.kt
package com.tonitealive.app.data

import com.tonitealive.app.domain.model.User
import rx.Observable

/**
 * A user cache
 */
interface UserCache {

    /**
     * Gets an {@link rx.Observable} that will emit a {@link User}
     *
     * @param username The username of the user to retrieve
     */
    fun get(username: String): Observable<User>

    /**
     * Puts an elements into the cache
     *
     * @param user The user to cache
     */
    fun put(user: User)

    /**
     * Checks if the user for the user ID is in the cache.
     *
     * @param username The username of the user
     *
     */
    fun isCached(username: String): Boolean

    /**
     * Checks if the cache is expired
     *
     * @return True if the cache is expired, false otherwise
     */
    fun isExpired(): Boolean

    /**
     * Evicts all items from the cache
     */
    fun evictAll()

}